app/src/main/kotlin/com/wire/android/ui/home/settings/account/handle/ChangeHandleViewModel.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.ui.home.settings.account.handle
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.android.gms.common.util.VisibleForTesting
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.ui.authentication.create.common.handle.HandleUpdateErrorState
+import com.wire.kalium.logic.feature.auth.ValidateUserHandleResult
+import com.wire.kalium.logic.feature.auth.ValidateUserHandleUseCase
+import com.wire.kalium.logic.feature.user.GetSelfUserUseCase
+import com.wire.kalium.logic.feature.user.SetUserHandleResult
+import com.wire.kalium.logic.feature.user.SetUserHandleUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class ChangeHandleViewModel @Inject constructor(
+    private val updateHandle: SetUserHandleUseCase,
+    private val validateHandle: ValidateUserHandleUseCase,
+    private val getSelf: GetSelfUserUseCase,
+    private val navigationManager: NavigationManager
+) : ViewModel() {
+
+    var state: ChangeHandleState by mutableStateOf(ChangeHandleState())
+        @VisibleForTesting
+        set
+
+    private var currentHandle: String? = null
+
+    init {
+        viewModelScope.launch {
+            getSelf().firstOrNull()?.handle.let {
+                currentHandle = it
+                state = state.copy(handle = TextFieldValue(it.orEmpty()))
+            }
+        }
+    }
```

Comment:
Can we pass the handle from previous screen ? this way this screen will flicker for short time, showing an empty value then the current handle

Reply:
i did test it and there is no flickering, and i am uncertain if retrieving values from  SavedStateHandle  is any faster than the local DB.

but we can run the init with runBlocking

Reply:
let's avoid runBlocking, it will block the current thread and could cause an ANR